File: analyze.py
import urllib.request
import re
from PIL import Image
from collections import Counter

# 1. Scrape the weblium template for hero image
url = "https://weblium.com/templates/demo/security-agency-website-design-131"
try:
    req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
    html = urllib.request.urlopen(req).read().decode('utf-8')
    
    # Look for background images or image tags
    images = re.findall(r'url\([\'"]?(https://[^)]+\.(?:jpg|jpeg|png|webp))[\'"]?\)', html)
    img_tags = re.findall(r'<img[^>]+src=[\'"](https://[^"\']+\.(?:jpg|jpeg|png|webp))[\'"]', html)
    
    print("Found background images:")
    for img in set(images):
        print("-", img)
        
    print("\nFound img tags:")
    for img in set(img_tags):
        print("-", img)
except Exception as e:
    print("Error fetching URL:", e)

# 2. Extract dominant color from logo
try:
    img = Image.open('public/logo.png')
    img = img.convert('RGB')
    
    # Get colors, excluding pure white/near white (background)
    pixels = img.getdata()
    valid_pixels = [p for p in pixels if not (p[0] > 240 and p[1] > 240 and p[2] > 240)]
    
    counts = Counter(valid_pixels)
    most_common = counts.most_common(5)
    
    print("\nDominant non-white colors in logo:")
    for color, count in most_common:
        print(f"- RGB: {color} (Hex: #{color[0]:02x}{color[1]:02x}{color[2]:02x}) Count: {count}")
        
except Exception as e:
    print("Error analyzing logo:", e)

File: analyze.js
const { Jimp } = require('jimp');

async function analyzeLogo() {
  try {
    const image = await Jimp.read('public/logo.png');
    const colorCounts = {};

    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      const r = this.bitmap.data[idx + 0];
      const g = this.bitmap.data[idx + 1];
      const b = this.bitmap.data[idx + 2];
      
      // Ignore near-white and near-black
      if (r > 240 && g > 240 && b > 240) return;
      if (r < 15 && g < 15 && b < 15) return;
      
      const hex = '#' + ((1 << 24) + (r << 16) + (g << 8) + b).toString(16).slice(1);
      colorCounts[hex] = (colorCounts[hex] || 0) + 1;
    });

    const sortedColors = Object.entries(colorCounts).sort((a, b) => b[1] - a[1]);
    console.log("Dominant non-white/black colors:", sortedColors.slice(0, 5));
  } catch (e) {
    console.error("Error:", e);
  }
}

analyzeLogo();
